docs/user_manual/dataset-terminology.md

```diff
@@ -10,23 +10,101 @@ Some terms regarding the data structures are explained here, including the defin
-    - **SingleArray** A dictionary where the keys are the component types and the values are one-dimensional structured numpy arrays.
-    - **BatchArray:** An array of dictionaries where the keys are the component types and the values are two-dimensional structured numpy arrays.
-        - **DenseBatchArray:** A two-dimensional structured numpy array containing a list of components of the same type for each scenario.
-        - **SparseBatchArray:** A dictionary with a one-dimensional numpy int64 array and a one-dimensional structured numpy arrays. 
-
-### Type of Dataset 
+```{mermaid}
+graph TD
+    subgraph Datasets
+    Dataset --> SingleDataset
+    Dataset --> BatchDataset
+    end
+
+    subgraph Dataset values
+    ComponentData --> DataArray
+    ComponentData --> ColumnarData
+
+    DataArray --> SingleArray
+    DataArray --> BatchArray
+
+    BatchArray --> DenseBatchArray
+    BatchArray --> SparseBatchArray
+
+    ColumnarData --> SingleColumnarData
+    ColumnarData --> BatchColumnarData
+
+    BatchColumnarData --> DenseColumnarData
+    BatchColumnarData --> SparseColumnarData
+    end
+
+    subgraph Other numpy arrays
+    Indptr
+    SingleColumn
+    BatchColumn
+    end
+
+    click Dataset href "../api_reference/python-api-reference.html#power_grid_model.data_types.Dataset"
+    click SingleDataset href "../api_reference/python-api-reference.html#power_grid_model.data_types.SingleDataset"
+    click BatchDataset href "../api_reference/python-api-reference.html#power_grid_model.data_types.BatchDataset"
+
+    click ComponentData href "../api_reference/python-api-reference.html#power_grid_model.data_types.ComponentData"
+    click DataArray href "../api_reference/python-api-reference.html#power_grid_model.data_types.DataArray"
+    click ColumnarData href "../api_reference/python-api-reference.html#power_grid_model.data_types.ColumnarData"
+    click SingleArray href "../api_reference/python-api-reference.html#power_grid_model.data_types.SingleArray"
+    click BatchArray href "../api_reference/python-api-reference.html#power_grid_model.data_types.BatchArray"
+    click DenseBatchArray href "../api_reference/python-api-reference.html#power_grid_model.data_types.DenseBatchArray"
+    click SparseBatchArray href "../api_reference/python-api-reference.html#power_grid_model.data_types.SparseBatchArray"
+    click SingleColumnarData href "../api_reference/python-api-reference.html#power_grid_model.data_types.SingleColumnarData"
+    click BatchColumnarData href "../api_reference/python-api-reference.html#power_grid_model.data_types.BatchColumnarData"
+    click DenseColumnarData href "../api_reference/python-api-reference.html#power_grid_model.data_types.DenseColumnarData"
+    click SparseColumnarData href "../api_reference/python-api-reference.html#power_grid_model.data_types.SparseColumnarData"
+
+    click Indexpointer href "../api_reference/python-api-reference.html#power_grid_model.data_types.Indexpointer"
```

Comment:
please merge with `main` so that you get the link to `Indexpointer` (or `IndexPointer`) if it exists